test/extended/util/cli.go

```diff
@@ -0,0 +1,225 @@
+package util
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	kclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	. "github.com/GoogleCloudPlatform/kubernetes/test/e2e"
+	. "github.com/onsi/ginkgo"
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/admin/policy"
+	"github.com/openshift/origin/pkg/cmd/cli"
+	"github.com/openshift/origin/pkg/cmd/server/bootstrappolicy"
+	testutil "github.com/openshift/origin/test/util"
+	"github.com/spf13/cobra"
+)
+
+// CLI provides function to call the OpenShift CLI and Kubernetes and OpenShift
+// REST clients.
+type CLI struct {
+	verb            string
+	adminConfigPath string
+	globalArgs      []string
+	commandArgs     []string
+	finalArgs       []string
+	stdout          io.Writer
+	verbose         bool
+	cmd             *cobra.Command
+	kubeFramework   *Framework
+}
+
+// NewCLI initialize the Kubernetes E2E framework, project role bindings.
+// If runInGinko argument is set to true, this function will install BeforeEach
+// hook to Gingo test case that grant project role bindings to a namespace created
+// by the E2E test. This option must be set inside Describe() or Context()
+func NewCLI(project, configPath string, runInGinko bool) *CLI {
+	client := &CLI{}
+	client.kubeFramework = NewFramework(project)
+	if len(configPath) == 0 {
+		FatalErr(fmt.Errorf("The configPath can't be empty"))
+	}
+	client.adminConfigPath = configPath
+	if runInGinko {
+		func() { BeforeEach(client.beforeEach) }()
+	} else {
+		client.SetNamespace(project)
+	}
+	return client
+}
+
+// OsFramework returns OpenShift framework
+func (c *CLI) OsFramework() *OsFramework {
+	return NewOsFramework(c.kubeFramework.Namespace, c.AdminRESTClient())
+}
+
+// KubeFramework returns Kubernetes framework
+func (c *CLI) KubeFramework() *Framework {
+	return c.kubeFramework
+}
+
+// SetNamespace overrides the namespace set by framework
+func (c *CLI) SetNamespace(ns string) {
```

Comment:
Make this
```go
func (c *CLI) SetNamespace(ns string) *CLI {
```

and above in `NewCLI`:
```go
if !runInGinko {
   return client.SetNamespace(project)
}
BeforeEach(client.beforeEach)
return client
```
Though I am not sure about the closure around `BeforeEach` (see comment above).